api/src/pcapi/core/users/external/zendesk.py

```diff
@@ -0,0 +1,266 @@
+) -> bool:
+    """
+    Update user with custom attributes
+    https://developer.zendesk.com/api-reference/ticketing/users/users/#update-user
+    """
+    data = {"user": format_contact_attributes(email, attributes)}
+
+    return _put_to_zendesk(zendesk_user_id, f"/users/{zendesk_user_id}", data)
+
+
+def _add_internal_note(
+    ticket_id: int, zendesk_user_id: int, email: str, attributes: Union[UserAttributes, ProAttributes]
+) -> bool:
+    """
+    Post an internal note so that support people can click on hyperlinks (not possible with custom attributes)
+    https://developer.zendesk.com/api-reference/ticketing/tickets/ticket_comments/
+    """
+    html_body = Markup("<i>Note automatique générée par le backend pass Culture ({})</i><br/>").format(settings.ENV)
+
+    if isinstance(attributes, ProAttributes):
+        if attributes.user_id:
+            html_body += Markup("Utilisateur pro identifié : <b>{} {}</b><br/>").format(
+                attributes.first_name, attributes.last_name
+            )
+            for bo_link in _get_backoffice_pro_user_links(email):
+                html_body += Markup('<a href="{}" target="_blank">{}</a><br/>').format(bo_link, bo_link)
+        if attributes.venue_ids:
+            venue_count = len(set(attributes.venue_ids))
+            html_body += f"{venue_count} lieux identifiés :" if venue_count > 1 else "1 lieu identifié :"
+            bo_link = _get_backoffice_venues_link(attributes.venue_ids)
+            html_body += Markup('<br/><a href="{}" target="_blank">{}</a><br/>').format(bo_link, bo_link)
+    else:
+        html_body += Markup("Utilisateur identifié : <b>{} {}</b>").format(attributes.first_name, attributes.last_name)
+        if attributes.date_of_birth:
+            html_body += f", {users_utils.get_age_from_birth_date(attributes.date_of_birth)} ans"
+        for bo_link in _get_backoffice_support_beneficiary_user_links(attributes.user_id):
+            html_body += Markup('<br/><a href="{}" target="_blank">{}</a>').format(bo_link, bo_link)
+
+    data = {
+        "ticket": {
+            "comment": {
+                "html_body": html_body,
+                "public": False,
+                "type": "Comment",
+            }
+        }
+    }
+
+    return _put_to_zendesk(zendesk_user_id, f"/tickets/{ticket_id}.json", data)
+
+
+def _put_to_zendesk(zendesk_user_id: int, route: str, data: dict) -> bool:
+    if settings.IS_RUNNING_TESTS:
+        testing.zendesk_requests.append({"zendesk_user_id": zendesk_user_id, "data": data})
+        return True
+
+    if settings.IS_DEV or not settings.ZENDESK_API_URL:
+        logger.info("A request to Zendesk API would be sent for user %s with data %s", zendesk_user_id, data)
+        return True
+
+    base64_authentication = base64.b64encode(
+        f"{settings.ZENDESK_API_EMAIL}/token:{settings.ZENDESK_API_TOKEN}".encode()
+    ).decode()
+
+    try:
+        response = requests.put(
+            f"{settings.ZENDESK_API_URL}{route}",
```

Comment:
Certains serveurs n'apprécient pas les double slashes. Est-ce qu'on ne se rajouterait pas une ceinture de sécurité ici (ou dans la définition dans `settings.py`) ?

```suggestion
            f"{settings.ZENDESK_API_URL.rstrip("/")}{route}",
```

Autre option : utiliser `urllib.parse.urljoin`. (Mais ça ne marchera pas si ZENDESK_API_URL contient un bout de chemin, comme par exemple "https://api.zendesk.com/v1".)